products.js

```diff
@@ -0,0 +1,308 @@
+//       rule(dom('span, div, li'), type('priceish')),
+//
+//       //bonus if direct text (not children) contains a dollar sign
+//       rule(type('priceish'), score(hasDollarSign)),
+//
+//       //look for good css within tag
+//       rule(type('priceish'), score(tagHasGoodCss)),
+//
+//       //not to be confused with amount off (minus sign, crossed off)
+//       rule(type('priceish'), score(notSavingsAmount)),
+//
+//       //return image with max score
+//       rule(type('priceish').max(), out('product-price'))
+//
+//     );
+//
+//     function contentFnodes(doc) {
+//         return rules.against(doc).get('product-price');
+//     }
+//
+//     return contentFnodes;
+// }
+
+/**
+ * Maintain state as we compare a series of DOMs, reporting the percent
+ * difference at the end.
+ */
+class DiffStats {
+    constructor(contentFnodes, feature) {
+        this.lengthOfExpectedTexts = 0;
+        this.lengthOfDiffs = 0;
+        this.feature = feature;
+        if (feature === 'image') {
+          this.contentFnodes = contentFnodes || tunedImageFnodes();
+        } else if (feature === 'title'){
+          this.contentFnodes = contentFnodes || tunedTitleFnodes();
+        } else {
+          this.contentFnodes = contentFnodes || tunedPriceFnodes();
+        }
+    }
+
+    compare(expectedDom, sourceDom) {
+        var expectedText;
+        var gotText;
+        if (this.feature === 'image') {
+          //compare images by src
+          expectedText = expectedDom.body.firstChild.src;
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.src)[0];
+        } else if (this.feature === 'title') {
+          //compare innerHTML text of titles
+          expectedText = expectedDom.head.firstChild.innerHTML;
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.innerHTML)[0];
+        } else if (this.feature === 'price') {
+          //strip whitespace and dollar sign if there is one when comparing price
+          expectedText = expectedDom.body.firstChild.textContent.replace('$', '').trim();
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.textContent.replace('$', '').trim())[0];
+        }
+
+        this.lengthOfExpectedTexts++;
+        if(expectedText !== gotText) {
+          this.lengthOfDiffs++;
+        }
+
+        // Uncomment for debugging:
+        console.log(leven(expectedText, gotText), expectedText.length, leven(expectedText, gotText)/expectedText.length);
+        console.log('Got:\n' + gotText);
+        console.log('\nExpected:\n' + expectedText);
+    }
+
+    score() {
+        return this.lengthOfDiffs / this.lengthOfExpectedTexts * 100;
+    }
+}
+
+function createDict(item, sourceDom){
+  var nodesDict = JSON.parse(readFileSync(join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', item, 'nodes.txt'), 'utf-8'));
+  var elems = sourceDom.getElementsByTagName('*');
```

Comment:
So, you're using `var` sometimes…and `let` and `const` other times. I feel like consistency here is a virtue. `let` and `const` are newer and let you control scope and indicate intent a bit better. That's why I used them in Fathom.